src/thunderbots/software/ai/intent/kick_intent.cpp

```diff
@@ -0,0 +1,24 @@
+#include "kick_intent.h"
+
+KickIntent::KickIntent(unsigned int robot_id, const Point &dest, const Angle &final_angle,
+                       double final_speed, int priority)
+    : KickPrimitive(robot_id, dest, final_angle, final_speed), priority(priority) 
+{
+}
+
+
+KickIntent::KickIntent(unsigned int robot_id, const Point &dest, const Angle &final_angle,
+                       double final_speed)
+    : KickPrimitive(robot_id, dest, final_angle, final_speed)
+{
+}
+
+std::string KickIntent::getIntentName() const
+{
+    return KICK_INTENT_NAME;
+}
+
+int KickIntent::getPriority() const
+{
+    return priority;
+}
```

Comment:
I think we can just give an implementation for this in the `Intent` class, that way we don't need to implement it (or specify that there is a `priority` class member) for each `Intent` subclass.

Reply:
I think `Intent` is meant to be an abstract class and each type of intent might have a different way of handling priority.

Reply:
Just had a quick conversation with @MathewMacDougall, looks like the intent here is to use the same priority for all intents, so we can probably just put this in the abstract class (note that the class will remain abstract because it still has virtual members, I can explain this further if ya like, just ping me on slack!)